ext/exif/exif.c

```diff
@@ -4420,7 +4424,8 @@ static bool exif_scan_HEIF_header(image_info_type *ImageInfo, unsigned char *buf
 				exif_isobmff_parse_meta(data, data + limit, &pos);
 			}
 			if ((pos.size) &&
-				(ImageInfo->FileSize >= pos.offset + pos.size) &&
+				(pos.size < ImageInfo->FileSize) &&
```

Comment:
Not related to your changes, but `pos` may be uninitialized here (even when `exif_isobmff_parse_meta` has been called)